src/compiler/scala/tools/nsc/ast/TreeInfo.scala

```diff
@@ -97,4 +97,12 @@ abstract class TreeInfo extends scala.reflect.internal.TreeInfo {
     case DocDef(_, definition) => isPureDef(definition)
     case _ => super.isPureDef(tree)
   }
+
+  override def firstConstructor(stats: List[Tree]): Tree = {
+    val unwrapped = stats map {
+      case DocDef(_, defn) => defn
```

Comment:
I don't know whether it's needed, but the above methods recurse to handle nested `DocDef`s -- should this one do the same?